src/System Application/App/SharePoint Authorization/src/SharePointClientCredentials.Codeunit.al

```diff
@@ -0,0 +1,73 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.Integration.Sharepoint;
+
+codeunit 9145 "SharePoint Client Credentials" implements "SharePoint Authorization"
+{
+    Access = Internal;
+    InherentPermissions = X;
+
+    var
+        ClientId: Text;
+        Certificate: Text;
+        AadTenantId: Text;
+        Scopes: List of [Text];
+        CertificatePassword: SecretText;
+
+    procedure SetParameters(NewAadTenantId: Text; NewClientId: Text; NewCertificate: Text; NewCertificatePassword: SecretText; NewScopes: List of [Text])
+    begin
+        AadTenantId := NewAadTenantId;
+        ClientId := NewClientId;
+        Certificate := NewCertificate;
+        CertificatePassword := NewCertificatePassword;
+        Scopes := NewScopes;
+    end;
+
+    procedure Authorize(var HttpRequestMessage: HttpRequestMessage);
+    var
+        Headers: HttpHeaders;
+        BearerTxt: Label 'Bearer %1', Comment = '%1 = Token', Locked = true;
+    begin
+        HttpRequestMessage.GetHeaders(Headers);
+        Headers.Add('Authorization', SecretStrSubstNo(BearerTxt, GetToken()));
+    end;
+
+    local procedure GetToken(): SecretText
+    var
+        ErrorText: Text;
+        AccessToken: SecretText;
+    begin
+        if not AcquireToken(AccessToken, ErrorText) then
+            Error(ErrorText);
+        exit(AccessToken);
+    end;
+
+    local procedure AcquireToken(var AccessToken: SecretText; var ErrorText: Text): Boolean
+    var
+        OAuth2: Codeunit System.Security.Authentication.OAuth2;
+        FailedErr: Label 'Failed to retrieve an access token.';
+        //TODO: Check Authority Url

```

Comment:
A todo in a non-draft PR? Shame! 😉😋

Reply:
I had problems setting up my new BC25 environment where I wanted to test this again.
I'll update this PR next week with some guideline on how to manually test this authentication method.

Reply:
I updated my code an added instructions on how to test the new authorization.